drill.cpp

```diff
@@ -388,16 +388,24 @@ bool ExcellonProcessor::millhole(std::ofstream &of, double start_x, double start
         stepcount = (int) ceil(abs(cutter->zwork / cutter->stepsize));
     }
 
+    double delta_x = stop_x - start_x;
+    double delta_y = stop_y - start_y;
+    double distance = sqrt(delta_x*delta_x + delta_y*delta_y);
     if (cutdiameter * 1.001 >= holediameter) { //In order to avoid a "zero radius arc" error
         // Hole is smaller than cutdiameter so just drill/zig-zag.
         of << "G0 X" << start_x * cfactor << " Y" << start_y * cfactor << '\n';
-        if (slot)
-        {
+        if (slot) {
+            double zhalfstep = cutter->zwork / stepcount / 2;
+            double feedrate;
+            if (abs(cutter->feed / distance) < abs(cutter->vertfeed / zhalfstep)) {
```

Comment:
What is the abs() for? In what case is this necessary?

Reply:
Not really, but it was in use above so I did it here, too. We can remove it and then just take the negative of zhalfstep. Sure.